Race condition in `make ghdllib -j8`

Here's a cleaned-up build log I had where `make ghdllib -j8` failed (compiling the `v0.36` branch):

```
forrest@forrest-tool:~/ghdl/work/ghdl$ make ghdllib -j8
...
mkdir grt
...
cd grt; MSYS2_ARG_CONV_EXCL="-aI;-gnatec" gnatmake -u -c -g -gnatec/home/forrest/ghdl/work/ghdl/./src/grt/grt.adc -gnatdY -o main.o /home/forrest/ghdl/work/ghdl/./src/grt/main.adb -Igrt
/bin/sh: 1: cd: can't cd to grt
...
ar rcv lib/ghdl/libgrt.a `sed -e "/^-/d" -e "s!^!grt/!" < grt/grt-files` \
 jumps.o times.o grt-cbinding.o grt-cvpi.o fstapi.o lz4.o fastlz.o grt/run-bind.o grt/main.o
...
ar: grt/main.o: No such file or directory
src/grt/Makefile.inc:109: recipe for target 'lib/ghdl/libgrt.a' failed
make: *** [lib/ghdl/libgrt.a] Error 1
```

It seems that `mkdir grt` hadn't completed by the time that `cd grt` was attempted. Because `cd grt` is followed by a semicolon and not `&&`, the failure was ignored, and gnatmake dropped `main.o` into the wrong directory. Later on, `ar` couldn't find `main.o`.

The race condition exists because the `grt/main.o` target doesn't have `grt` (which runs `mkdir grt`) as a prerequisite, so `mkdir grt` runs in parallel with the `cd grt` command.

This still appears to be broken in the `master` branch (by inspection; I haven't been able to replicate this.)

Attached pull request #1260 fixes this.

--- a/src/grt/Makefile.inc
+++ b/src/grt/Makefile.inc
@@ -139,21 +139,21 @@ $(libdirsuffix)/libgrt.a: $(GRT_ADD_OBJS) grt/run-bind.o grt/main.o grt/grt-file
 	$(GRT_RANLIB) $@
 
 grt/run-bind.adb: grt $(GRT_SRC_DEPS) grt-force
-	cd grt; $(MSYS2_ARG) $(GNATMAKE) -c -aI$(GRTSRCDIR) -aI.. \
+	cd grt && $(MSYS2_ARG) $(GNATMAKE) -c -aI$(GRTSRCDIR) -aI.. \
 	  $(GRT_PRAGMA_FLAG) ghdl_main $(GRT_ADAFLAGS) -cargs $(GRT_FLAGS)
 # 	Set No Run-Time flag to suppress references to standard gnat library.
 	sed -e '/^P /s/P /P NR /' < grt/ghdl_main.ali > grt/ghdl_main-tmp.ali
 	$(MV) grt/ghdl_main-tmp.ali grt/ghdl_main.ali
-	cd grt; $(GNATMAKE) -b ghdl_main.ali -bargs  -Lgrt_ -o run-bind.adb -n
+	cd grt && $(GNATMAKE) -b ghdl_main.ali -bargs  -Lgrt_ -o run-bind.adb -n
 
 grt:
 	mkdir grt
 
-grt/run-bind.o: grt/run-bind.adb
-	cd grt; $(MSYS2_ARG) $(GRT_ADACOMPILE) -o run-bind.o run-bind.adb
+grt/run-bind.o: grt grt/run-bind.adb
+	cd grt && $(MSYS2_ARG) $(GRT_ADACOMPILE) -o run-bind.o run-bind.adb
 
-grt/main.o: $(GRTSRCDIR)/main.adb
-	cd grt; $(MSYS2_ARG) $(GRT_ADACOMPILE) -o main.o $< -Igrt
+grt/main.o: grt $(GRTSRCDIR)/main.adb
+	cd grt && $(MSYS2_ARG) $(GRT_ADACOMPILE) -o main.o $(GRTSRCDIR)/main.adb -Igrt
 
 GRT_C_COMPILE = $(CC) -c $(GRT_FLAGS) -o $@ $<
 GRT_C_COMPILE_PIC = $(CC) -c $(GRT_FLAGS) $(PIC_FLAGS) -o $@ $<

